lib/proxy_api/external_ipam.rb

```diff
@@ -44,30 +44,28 @@ def initialize(args)
     #   }
     # }
     #
-    # Params: 1. subnet:  The IPv4 or IPv6 subnet CIDR. (Examples: IPv4 - "100.10.10.0/24", IPv6 - "2001:db8:abcd:12::/124")
-    #         2. mac:     The mac address of the interface obtaining the IP address for
-    #         3. group:   The group in External IPAM that the subnet belongs to(e.g. This would be "Section" in phpIPAM)
+    # Params: 1. subnet:           The IPv4 or IPv6 subnet CIDR. (Examples: IPv4 - "100.10.10.0/24", IPv6 - "2001:db8:abcd:12::/124")
+    #         2. mac:              The mac address of the interface obtaining the IP address for
+    #         3. group(optional):  The group in External IPAM that the subnet belongs to (e.g. 'Section' in
+    #                              in phpIPAM, or 'VRF' in Netbox)
     #
-    # Returns: A hash with the next available IP in the "data" field, or a hash with "error" key
-    #          containing error message
-    #
-    # Examples:
+    # Responses:
     #   Responses if success:
-    #     IPv4: {"data": "100.55.55.3"}
-    #     IPv6: {"data": "2001:db8:abcd:12::1"}
+    #     "100.55.55.3"          (IPv4)
+    #     "2001:db8:abcd:12::1"  (IPv6)
     #   Response if missing required params:
     #     {"error": ["A 'cidr' parameter for the subnet must be provided(e.g. 100.10.10.0/24)","A 'mac' address must be provided(e.g. 00:0a:95:9d:68:10)"]}
     #   Response if subnet does not exist:
-    #     {"error": "The specified subnet does not exist in External IPAM."}
+    #     {"error": "Subnet not found in External IPAM."}
     #   Response if there are no free addresses:
     #     {"error": "No free addresses found"}
     #   Response if can't connect to External IPAM server
     #     {"error": "Unable to connect to External IPAM server"}
     def next_ip(subnet, mac, group = "")
       raise "subnet cannot be nil" if subnet.nil?
+      raise "mac address cannot be nil" if mac.nil?
       response = parse get("/subnet/#{subnet}/next_ip?mac=#{mac}&group=#{URI.escape(group.to_s)}")
-      raise(response['error']) if response['error'].present?
-      response['data']
+      process(response)
```

Comment:
This method should actually expect the "data" element and clean it out for caller. The code was better before.